inc/authsaml.class.php

```diff
@@ -0,0 +1,393 @@
+         echo "</td>";
+         echo "</td>";
+         echo "<td>" . __('Fingerprint algorithm'). "</td>";
+         echo "<td>";
+         $elements = array('sha1', 'sha256', 'sha384', 'sha512');
+         Dropdown::showFromArray(
+                 'idp_certfingerprintalgorithm',
+                 $elements,
+                 array('value' => $this->fields["idp_certfingerprintalgorithm"]));
+         echo "</td>";
+         echo "</tr>";
+
+         echo "<tr>";
+         echo "<th colspan='2'>" . __('SSO endpoint info (for login)') . "</th>";
+         echo "<th colspan='2'>" . __('SSO endpoint info (for logout)') . "</th>";
+         echo "</tr>";
+
+         echo "<tr class='tab_bg_1'>";
+         echo "<td>" . __('URL Location'). "</td>";
+         echo "<td>";
+         echo "<input type='text' name='idp_singlesignonservice_url' size='60' value=\"".$this->fields["idp_singlesignonservice_url"]."\">";
+         echo "</td>";
+         echo "<td>" . __('URL Location'). "</td>";
+         echo "<td>";
+         echo "<input type='text' name='idp_singlelogoutservice_url' size='60' value=\"".$this->fields["idp_singlelogoutservice_url"]."\">";
+         echo "</td>";
+         echo "</tr>";
+
+         echo "<tr class='tab_bg_1'>";
+         echo "<td>" . __('Protocol binding'). "</td>";
+         echo "<td>";
+         echo "<input type='text' name='idp_singlesignonservice_binding' size='60' value=\"".$this->fields["idp_singlesignonservice_binding"]."\">";
+         echo "</td>";
+         echo "<td>" . __('Protocol binding'). "</td>";
+         echo "<td>";
+         echo "<input type='text' name='idp_singlelogoutservice_binding' size='60' value=\"".$this->fields["idp_singlelogoutservice_binding"]."\">";
+         echo "</td>";
+         echo "</tr>";
+
+         $this->showFormButtons($options);
+
+      } else {
+         echo "<div class='center'>&nbsp;<table class='tab_cadre_fixe'>";
+         echo "<tr><th colspan='2'>" . self::getTypeName(1) . "</th></tr>";
+         echo "<tr class='tab_bg_2'><td class='center'>";
+         echo "<p class='red'>". __("The Openssl extension of your PHP parser isn't installed")."</p>";
+         echo "<p>".__('Impossible to use Openssl as external source of connection')."</p>".
+              "</td></tr></table></div>";
+      }
+
+      echo "<div align='center'>";
+      echo "<a href='".$this->getFormURL()."?metadata' target='_blank' class='vsubmit'>".__('Generate SAML metadata (XML file)')."<a>";
+      echo "</div><br/><br/>";
+   }
+
+
+
+   function generate_settings() {
+      $this->getFromDB(1);
+
+      $sp_x509cert = $this->fields['sp_x509cert'];
+      $sp_x509cert = str_replace('-----BEGIN CERTIFICATE-----', '', $sp_x509cert);
+      $sp_x509cert = str_replace('-----END CERTIFICATE-----', '', $sp_x509cert);
+      $sp_x509cert = str_replace(array("\r\n", "\r", "\n"), "", $sp_x509cert);
+      $sp_x509cert = trim($sp_x509cert);
+
+$privkey = "-----BEGIN RSA PRIVATE KEY-----
```

Comment:
Where is key from ?
Should probably be store in a more secure way (config dir) and generated on install.